jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java

```diff
@@ -139,7 +147,14 @@
 
     if (containerConfigurationTemplate.getContainerEnvironment() != null) {
       for (String environmentVariable : containerConfigurationTemplate.getContainerEnvironment()) {
-        imageBuilder.addEnvironmentVariableDefinition(environmentVariable);
+        Matcher matcher = environmentPattern.matcher(environmentVariable);
+        if (!matcher.matches()) {
+          throw new BadContainerConfigurationFormatException(
+              "Invalid environment variable definition: " + environmentVariable);
+        }
+        String name = matcher.group(1);
+        String value = matcher.group(2);
```

Comment:
When patterns are far away from their use, it can help to use a named group (e.g., `matcher.group("name")` and `matcher.group("value")`).

Reply:
+1 We should probably do this for some of the other regex patterns too.